src/Language/PureScript/CoreFn/Optimizer/PassThroughCases.hs

```diff
@@ -0,0 +1,81 @@
+-- |
+-- Optimizer step for simplifying "pass-through cases", for example:
```

Comment:
Could you please explain here why this optimization is not possible elsewhere (types) and why it is safe?